mzLib/Omics/BioPolymerWithSetModsExtensions.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System.Text;
+using Chemistry;
+using Omics.Modifications;
+
+namespace Omics;
+
+public static class BioPolymerWithSetModsExtensions
+{
+    public static string FullSequenceWithMassShift(this IBioPolymerWithSetMods withSetMods)
+    {
+        var subsequence = new StringBuilder();
+
+        // modification on peptide N-terminus
+        if (withSetMods.AllModsOneIsNterminus.TryGetValue(1, out Modification mod))
+        {
+            subsequence.Append('[' + mod.MonoisotopicMass.RoundedDouble(6).ToString() + ']');
+        }
+
+        for (int r = 0; r < withSetMods.Length; r++)
+        {
+            subsequence.Append(withSetMods[r]);
+
+            // modification on this residue
+            if (withSetMods.AllModsOneIsNterminus.TryGetValue(r + 2, out mod))
+            {
+                if (mod.MonoisotopicMass > 0)
+                {
+                    subsequence.Append("[+" + mod.MonoisotopicMass.RoundedDouble(6).ToString() + ']');
+                }
+                else
+                {
+                    subsequence.Append("[" + mod.MonoisotopicMass.RoundedDouble(6).ToString() + ']');
+                }
+            }
+        }
+
+        // modification on peptide C-terminus
+        if (withSetMods.AllModsOneIsNterminus.TryGetValue(withSetMods.Length + 2, out mod))
+        {
+            if (mod.MonoisotopicMass > 0)
+            {
+                subsequence.Append("[+" + mod.MonoisotopicMass.RoundedDouble(6).ToString() + ']');
+            }
+            else
+            {
+                subsequence.Append("[" + mod.MonoisotopicMass.RoundedDouble(6).ToString() + ']');
+            }
+        }
+        return subsequence.ToString();
+    }
+
+    /// <summary>
+    /// This method returns the full sequence with mass shifts INSTEAD OF PTMs in brackets []
+    /// Some external tools cannot parse PTMs, instead requiring a numerical input indicating the mass of a PTM in brackets
+    /// after the position of that modification
+    /// N-terminal mas shifts are in brackets prior to the first amino acid and apparently missing the + sign
+    /// </summary>
+    /// <returns></returns>
+    public static string EssentialSequence(this IBioPolymerWithSetMods withSetMods,
+        IReadOnlyDictionary<string, int> modstoWritePruned)
+    {
```

Comment:
I can't see where mass shifts get written in this method, it looks like it just returns the standard mod formula. Is the summary comment incorrect?

Reply:
I think this summary comment was supposed to go one function up. It would be nice to have summary comments for all the functions in this class

Reply:
Fixed this and added new summary comment to EssentialSequence and DetermineFullSequence